File: py/Power_of_Three.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n <= 0 or n > 1162261467:
            return False
        while n != 1:
            if n % 3 != 0:
                return False
            n //= 3
        return  n == 1

if __name__ == '__main__':
    s = Solution()
    print(s.isPowerOfThree(9))

File: c_plus/Power_of_Three.cpp
#include<iostream>
using namespace std;

class Solution {
public:
    bool isPowerOfThree(int n) {
        if (n <= 0 or n > 1162261467) return false;
        while (n != 1) {
            if (n % 3 != 0) return false;
            n /= 3;
        }
        return n == 1;
    }
};

int main() {
    Solution s;
    cout << s.isPowerOfThree(45) << endl;
}
